lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -1,6 +1,25 @@
 defmodule LivebookWeb.SessionLive.SecretsComponent do
   use LivebookWeb, :live_component
 
+  @impl true
+  def mount(socket) do
+    {:ok, assign(socket, initialized: false)}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    socket = assign(socket, assigns)
+
+    socket =
+      if socket.assigns.initialized do
+        socket
+      else
+        assign(socket, data: %{"label" => assigns.secret, "value" => ""}, initialized: true)
```

Comment:
Also, maybe it would be more clear if the assign we pass to the component was something like `initial_secret_label` (so it's clear that it's used only once)?

Reply:
I agree. I just think that the `secret_label` would be enough, since it's a form. 🤔 Wdyt?

Reply:
I'm fine either way! Theoretically the form state could be outside the component and be passed down, so `initial_` emphasizes that it's only the initial label, not the currently entered one.